### ISSUE TITLE
Question. Best way to set title bar hidden after app start

### ISSUE THREAD

**Issue by AndriiHarashchak** [leanflutter/window_manager#115]
Hello author. I have a problem. I want the app to be shown on start (default behavior) without `WaitUntilWindowReadyToShow.then(_)` (i have some troubles that the window is not always properly paints by another library when using this). When i try to call it in `main` `windowManager.setTitleBarStyle('hidden')`  i reseive "not responding" error. When I call it somewhere after the app is rendered it works well. So. What is the best place to call this instead of main (or in main but with some trick)?

**Comment by damywise** [comment]
`setTitleBarStyle('hidden')` is no longer used, please upgrade to version 0.2.1 and change the code to `setTitleBarStyle(TitleBarStyle.hidden)`

**Comment by AndriiHarashchak** [comment]
Enums (and maby some inner changes you have made) have not changed the described behavior. So to let this thing work I found a nice solution. For main stateful widget (some page that renders after start )  in `initState` I added `PostFrameCallback` in which I call `windowManager.setTitleBarStyle`. This solution helps to prevent "not responding" Windows error.

**Comment by damywise** [comment]
Interesting. I have encountered this "not responding" error before, which happens only when `setTitlebarstyle` is called. I'll try to remember what I did wrong and how I fixed it. Have you tried `flutter clean` and is the project made with the latest Flutter SDK version?

**Comment by damywise** [comment]
Also,
May I know what is this other library you are using? 
Is the "not responding" error still there if you use the window_manager only, without the other library?

**Comment by damywise** [comment]
I found something.
In my case, the app won't crash if:
- Calling window manager functions before calling `runApp` without using `await`
```dart
  WidgetsFlutterBinding.ensureInitialized();
  await windowManager.ensureInitialized();
  windowManager.setSize(const Size(1000, 500));
  runApp(const MyApp());
```
- Calling window manager functions after calling `runApp`
```dart
  WidgetsFlutterBinding.ensureInitialized();
  await windowManager.ensureInitialized();
  runApp(const MyApp());
  await windowManager.setSize(const Size(1000, 500));
```
- Calling window manager functions on `WidgetsBinding.addPostFrameCallback`
```dart
  WidgetsBinding widgetsBinding = WidgetsFlutterBinding.ensureInitialized();
  await windowManager.ensureInitialized();
  widgetsBinding.addPostFrameCallback((_) {
    await windowManager.setSize(const Size(1000, 500));
  });
  runApp(const MyApp());
```
- Calling window manager functions on `waitUntilReadyToShow`
```dart
  WidgetsFlutterBinding.ensureInitialized();
  await windowManager.ensureInitialized();
  windowManager.waitUntilReadyToShow().then((_) async {
    await windowManager.setSize(const Size(1000, 500));
  });
  runApp(const MyApp());
```
So, I guess, do one of those that works for you.

**Comment by AndriiHarashchak** [comment]
Thank you for your reply. I see this too. (I spent a few days setting up my app). And this works as you described. But instead of one function. `setTitleBarStyle(TitleBarStyle someStyle)` not working when calling it in main. only in `WaitUntilWindowsReadyToShow` or in `PostFrameCallback`. So that`s why I opened this issue to ask if any interesting solutions are available. Anyway, you showed all of them. So I think this issue can be closed. Thanks.
